code/game/objects/items/kirbyplants.dm

```diff
@@ -26,10 +25,6 @@
 	AddComponent(/datum/component/two_handed, require_twohands=TRUE, force_unwielded=10, force_wielded=10)
 	AddElement(/datum/element/beauty, 500)
 
-/obj/item/kirbyplants/update_icon_state()
-	. = ..()
-	icon_state = dead ? "plant-25" : base_icon_state
```

Comment:
update_icon is being called on throw because it has the two_handed component.
_______________________________

This isn't the correct fix. Revert your changes.


```
/obj/item/kirbyplants/Initialize(mapload)
	. = ..()
	AddComponent(/datum/component/tactical)
	AddComponent(/datum/component/two_handed, require_twohands=TRUE, force_unwielded=10, force_wielded=10)
	AddElement(/datum/element/beauty, 500)
```
need to add 
```
	if(icon_state != base_icon_state && icon_state != "plant-25")
		base_icon_state = icon_state 
```
to the end of init for map varedit support.

_____________________________


The issue is that none of these are updating the `base_icon_state` var.
```
/obj/item/kirbyplants/random/Initialize(mapload)
	. = ..()
	icon = 'icons/obj/flora/plants.dmi'
	if(!random_plant_states)
		generate_states()
	icon_state = pick(random_plant_states)
```
should be 
```
base_icon_state = pick(random_plant_states)
update_appearance(UPDATE_ICON)
```

and this one
```
/obj/item/kirbyplants/proc/change_visual()
	if(!random_plant_states)
		generate_states()
	var/current = random_plant_states.Find(icon_state)
	var/next = WRAP(current+1,1,length(random_plant_states))
	icon_state = random_plant_states[next]
```
should be
```
base_icon_state = random_plant_states[next]
update_appearance(UPDATE_ICON)
```